code/src/components/ClearButton.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { tasks } from 'reducers/tasks'
+import 'css/buttons.css'
+
+export const ClearButton = () => {
+  const dispatch = useDispatch()
+  const allTasks = useSelector((state) => state.tasks)
```

Comment:
Since you're only using the length of the array, I'd recommend returning that directly from the selector:

```
const allTasks = useSelector(state => state.tasks.length)
When you do it like you have, this ClearButton component will re-render whenever the array of tasks changes (for example toggling a task's completion), but if you return the length from the selector, it'll only re-render when the length of the array changes. Small optimisation, but good to keep in the back of your head if you're working on a bigger project where you want to avoid unnecessary re-renders.
```